server/src/main/java/org/elasticsearch/cluster/metadata/RollupIndexMetadata.java

```diff
@@ -0,0 +1,162 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.cluster.metadata;
+
+import org.elasticsearch.cluster.AbstractDiffable;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.time.WriteableZoneId;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.LoggingDeprecationHandler;
+import org.elasticsearch.common.xcontent.NamedXContentRegistry;
+import org.elasticsearch.common.xcontent.ObjectParser;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.search.aggregations.bucket.histogram.DateHistogramInterval;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.optionalConstructorArg;
+
+/**
+ * Metadata about rollup indices that includes information about their capabilities,
+ * such as which date-intervals and date-timezones they are configured with and what metric aggregations
+ * do they support.
+ *
+ * The information in this class will be used to decide which index within a rollup group will be chosen
+ * for a specific aggregation. For example, if there are two indices with different intervals (`1h`, `1d`) and
+ * a date-histogram aggregation request is sent for daily intervals, then the index with the associated `1d` interval
+ * will be chosen.
+ */
+public class RollupIndexMetadata extends AbstractDiffable<RollupIndexMetadata> implements ToXContentObject {
+
+    public static final String TYPE = "rollup";
+    public static final String SOURCE_INDEX_NAME_META_FIELD = "source_index";
+    public static final String ROLLUP_META_FIELD = "meta";
+
+    private static final ParseField DATE_INTERVAL_FIELD = new ParseField("interval");
+    private static final ParseField DATE_TIMEZONE_FIELD = new ParseField("timezone");
+    private static final ParseField METRICS_FIELD = new ParseField("metrics");
+
+    // the date interval used for rolling up data
+    private final DateHistogramInterval dateInterval;
+    // the timezone used for the date_histogram
+    private final WriteableZoneId dateTimezone;
+    // a map from field name to metrics supported by the rollup for this field
+    private final Map<String, List<String>> metrics;
+
```

Comment:
I think we should move all these informations in the mapping. The interval and timezone could be metadata associated with the mapped field and metrics are already part of the mapping. The only useful information here would be the source index and the marker that this index is a rollup. The rest should be part of the mapping imo.

Reply:
Since you started in the direction above, can you remove this metadata all together ? No need to introduce something that we don't plan to use.

Reply:
This is complete now that #69921 has been merged.

All field related information have moved to the field metadata. Index metadata contains only the `source.index_name` and `source.index_uuid`